rad/src/lib.rs

```diff
@@ -81,19 +85,40 @@ pub fn try_data_request(
                 .unwrap_or_else(|error| RadonReport::from_result(Err(error), &retrieval_context))
         })
         .collect();
-    let retrieval_values: Vec<RadonTypes> = retrieval_reports
-        .iter()
-        .map(|report| report.result.clone())
-        .collect();
 
-    let (aggregation_result, aggregation_context) = run_aggregation_report(
-        retrieval_values,
-        &request.aggregate,
-        settings,
-        current_active_wips(),
+    // Evaluate aggregation pre-condition by using the same logic than for tally pre-condition,
```

Comment:
This is duplicated in `rad/src/lib.rs` and in `node/src/actors/rad_manager/handlers.rs`. It would be nice to move this logic into a new function, similar to `run_tally` which handles the precondition internally.

Reply:
Actually I just noticed that there are two different `run_tally` functions, one in `rad/src/lib.rs` and one in `validations/src/validations.rs`. The one from validations handles the preconditions and calls the one from rad...

So this `try_data_request` function is missing the tally precondition check, but it may be not needed because there should always be only one "reveal", so it's always in consensus with itself, right?